stub/src/main/java/io/grpc/stub/ServerCallStreamObserver.java

```diff
@@ -143,4 +145,28 @@ public void disableAutoRequest() {
    */
   @Override
   public abstract void setMessageCompression(boolean enable);
+
+  /**
+   * Sets a {@link Runnable} to be executed when the call is closed correctly from the server's
+   * point of view: either {@link #onCompleted()} or {@link #onError(Throwable)} has been called,
+   * all the messages and trailing metadata have been put on the wire and the stream has been
+   * closed.
+   * Note however that the client still may have not received all the messages due to race
+   * conditions or crashes.
+   *
+   * <p>Unless server exits abruptly, if both {@code onCloseHandler} and {@code onCancelHandler}
+   * are set, then after a call to either {@link #onCompleted()} or {@link #onError(Throwable)}
+   * exactly 1 of {@code onCloseHandler} and {@code onCancelHandler} will be called eventually.</p>
```

Comment:
Spell out "one."

"eventually" doesn't help the user too much, and it is technically incorrect if keepalive isn't enabled. And this sort of places some wrong emphasis of onCancel's relationship with onCompleted/onError. Maybe refocusing it:

> `Exactly one of {@code onCloseHandler} and {@code onCancelHandler} is guaranteed to be called when the RPC terminates.`

Then we just rely on the earlier paragraph that requires calling onCompleted/onError for onCloseHandler to be called.

Reply:
done